src/main/java/seedu/duke/model/ContainerList.java

```diff
@@ -1,46 +1,140 @@
 package seedu.duke.model;
 
+import seedu.duke.model.exception.DuplicateItemContainerException;
 import seedu.duke.model.exception.IllegalArgumentException;
+import seedu.duke.model.exception.ItemContainerNotExistException;
+import seedu.duke.model.exception.ItemNotExistException;
 
 import java.util.ArrayList;
 
 /**
  * Represents a collection of ItemContainers.
+ * Provides methods to better manage ItemContainers.
+ * Implemented using Singleton pattern.
  */
 public class ContainerList {
-    int numOfContainers;
+
+    private static ContainerList containerList;
     ArrayList<ItemContainer> containers;
 
-    ContainerList() {
-        numOfContainers = 0;
+    private ContainerList() {
         containers = new ArrayList<ItemContainer>();
     }
 
+    public static ContainerList getContainerList() {
+        if (containerList == null) {
+            containerList = new ContainerList();
+        }
+        return containerList;
+    }
+
+    public void resetContainerList() {
+        containers = new ArrayList<>();
+    }
+
+    /**
+     * Adds the specified itemContainer to the ContainerList.
+     * (not recommended, as this method skips some checks)
+     *
+     * @param itemContainer The ItemContainer to be added
+     */
+    public void addContainer(ItemContainer itemContainer) {
+        containers.add(itemContainer);
+    }
+
     /**
-     * Creates a new ItemContainer with the name of the location in the ContainerList.
+     * Creates a new ItemContainer with the specified name in the ContainerList.
      *
-     * @param location A brief description of the position of ItemContainer
+     * @param name The name of the new ItemContainer
+     * @throws IllegalArgumentException        if name does not follow the format
+     * @throws DuplicateItemContainerException if there already exist a ItemContainer with this name
      */
-    public void addContainer(String location) throws IllegalArgumentException {
-        containers.add(new ItemContainer(location));
+    public ItemContainer addContainer(String name) throws IllegalArgumentException, DuplicateItemContainerException {
+        if (existContainer(name)) {
+            throw new DuplicateItemContainerException(name);
+        }
+
+        ItemContainer temp = new ItemContainer(name);
+        containers.add(temp);
+        return temp;
     }
 
-    // todo add deleteContainer method
+    /**
+     * Remove the item container from the ContainerList.
+     *
+     * @param itemContainer The ItemContainer to be removed
+     */
+    public void deleteContainer(ItemContainer itemContainer) {
+        assert containers.contains(itemContainer);
```

Comment:
Why use assert here instead of an Exception (SelectedItemContainerNotFound, etc)

Reply:
By right all item container are in ContainerList. but true I should have assert !containers.contains(itemContainer) at the end of the program. Will change it.